jetty-core/jetty-io/src/test/java/org/eclipse/jetty/io/ArrayByteBufferPoolTest.java

```diff
@@ -73,7 +72,7 @@ public void testMaxMemoryEviction()
         assertThat(pool.getDirectByteBufferCount(), greaterThan(0L));
         assertThat(pool.getDirectByteBufferCount(), lessThan((long)buffers.size()));
         assertThat(pool.getDirectMemory(), greaterThan(0L));
-        assertThat(pool.getDirectMemory(), lessThanOrEqualTo(40L));
+        assertThat(pool.getDirectMemory(), lessThan(120L));
```

Comment:
This seems to indicate that we can now badly overshoot. I'm wondering if the configurable memory limit makes any more sense given this.

Reply:
We can overshoot a bit for a while, but I don't think it is badly, as the next thread to evict will again attempt to get it down all the way.
This was just a lazy test change and we really need to do a probabilistic test to match the algorithm

Reply:
@lorban I've improved the test a bit.  We still overshoot, but you can see it is not by much